Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +166,180 @@ public static class CustomEntitySearch
+                    }
+                    else
+                    {
+                        escapedWord.Append(word[currWordCharIndex]);
+                    }
+                }
+                if (!word.Last().IsDelineating() && !substringMatch)
+                    escapedWord.Append(@"\b");
+                escapedWord.Append("))");
+                string pattern = (caseSensitive) ? @"(?x)" + escapedWord : @"(?ix)" + escapedWord;
+                
+                MatchCollection entityMatch = Regex.Matches(text, pattern, RegexOptions.Compiled, TimeSpan.FromSeconds(MaxRegexEvalTime));
+                if (entityMatch.Count != 0)
+                {
+                    foreach (Match match in entityMatch)
+                    {
+                        entities.Add(
+                            new Entity
+                            {
+                                Category = "customEntity",
+                                Value = match.Groups[1].Value,
+                                Offset = match.Index,
+                                Confidence = 0
+                            });
+                    }
+                    entitiesFound.Add(checkMatch);
+                }
+            }
+            else
+            {
+                List<int> startPointersInText = new List<int> { 0 };
+                List<int> endPointersInText = new List<int>();
+                string textCharArray = (caseSensitive) ? CreateWordArray(text) : CreateWordArray(text.ToLower(CultureInfo.CurrentCulture));
+                for (int currTextCharIndex = 0; currTextCharIndex < textCharArray.Length; currTextCharIndex++)
+                {
+                    if (textCharArray[currTextCharIndex].IsDelineating())
+                    {
+                        if (currTextCharIndex + 1 < textCharArray.Length && !textCharArray[currTextCharIndex + 1].IsDelineating())
+                            startPointersInText.Add(currTextCharIndex + 1);
+                        if (currTextCharIndex - 1 >= 0 && !textCharArray[currTextCharIndex - 1].IsDelineating())
+                            endPointersInText.Add(currTextCharIndex - 1);
+                    }
+                }
+                endPointersInText.Add(textCharArray.Length - 1);
+
+                double[] minLevenshteinDistance = new double[startPointersInText.Count];
+                int[] endofMatchInTextPointer = new int[startPointersInText.Count];
+                for (int startPointerIndex = 0; startPointerIndex < startPointersInText.Count; startPointerIndex++)
+                {
+                    minLevenshteinDistance[startPointerIndex] = leniency + 1;
+                    for (int endPointerIndex = startPointerIndex; endPointerIndex < endPointersInText.Count; endPointerIndex++)
+                    {
+                        if (endPointersInText[endPointerIndex] - startPointersInText[startPointerIndex] + 1 > checkMatch.Length * 2) break;
```

Comment:
>endPointersInText[endPointerIndex] - startPointersInText[startPointerIndex] + 1 [](start = 28, length = 79)

nit: this expression is used twice. Might be nice to store in tmp variable for readability (eg: potentialMatchLength